LayoutTests/editing/selection/301539.html

```diff
@@ -0,0 +1,91 @@
+<!doctype html>
+  <meta charset="UTF-8">
+<title>Testing bug #301539</title>
+<script src="../../resources/js-test.js"></script>
+
+<script>
+    jsTestIsAsync = true;
+
+    function onNextFrame(callback) {
+        return new Promise((resolve, reject) => {
+            requestAnimationFrame(() => {
+                callback();
+                resolve();
+            });
+        });
+    }
+
+   async function init() {
+       const iframe = document.querySelector("iframe");
+
+       await onNextFrame(() => {
+            document.getElementById("input").focus();
+            document.getElementById("input").select();
+       });
+
+       await onNextFrame(() => {
+            const fullScrollHeight = document.body.scrollHeight - window.innerHeight;
+            window.scrollTo(0, fullScrollHeight);
+       });
+
+       await onNextFrame(() => {
+            const target = iframe.contentDocument.getElementById("input");
+            target.select();
+        });
+
+       await onNextFrame(() => {
+            shouldBeTrue("window.scrollY === (document.body.scrollHeight - window.innerHeight)");
+            finishJSTest();
+        });
+
+    }
+</script>
+<body onload="init()">
+    <input id="input" value="birthday">
+
+<!-- some content to make the page scrollable -->
```

Comment:
You can achieve this with `input { margin-bottom: some-large-value; }` to keep the test output simpler.